analyzers/src/SonarAnalyzer.CSharp/Extensions/AssignmentExpressionSyntaxExtensions.cs

```diff
@@ -26,8 +26,114 @@
 
 namespace SonarAnalyzer.Extensions
 {
-    public static class AssignmentExpressionSyntaxExtensions
+    internal static class AssignmentExpressionSyntaxExtensions
     {
+        public readonly record struct AssignmentMapping(CSharpSyntaxNode Left, ExpressionSyntax Right);
+
+        /// <summary>
+        /// Maps the left and the right side arguments of an <paramref name="assignment"/>. If both sides are tuples, the tuple elements are mapped.
+        /// <code>
+        /// (var x, var y) = (1, 2);                 // [x→1, y→2]
+        /// var (x, y) = (1, 2);                     // [x→1, y→2]
+        /// (var x, (var y, var z)) = (1, (2, 3));   // [x→1, y→2, z→3]
+        /// var x = 1;                               // [x→1]
+        /// (var x, var y) = M();                    // [(x,y)→M()]
+        /// </code>
+        /// </summary>
+        /// <param name="assignment">The <paramref name="assignment"/> expression.</param>
+        /// <returns>A mapping from expressions on the left side of the <paramref name="assignment"/> to the right side.</returns>
```

Comment:
While the `<summary>` is useful here, the parameter and return type are just mechanical noise.
```suggestion
```